File: syspzl-api/src/repositories/endereco.repository.js
import pool from '../database/database.connection.js';

class EnderecoRepository {
  // Busca um endereço pelo ID do AUTUADO, que é o mais comum
  async findByAutuadoId(autuadoId) {
    const [rows] = await pool.query('SELECT * FROM `endereco` WHERE autuado_id = ?', [autuadoId]);
    return rows[0];
  }

  async create(enderecoData) {
    const { autuado_id, estado, cidade, cep, bairro, logradouro, numero, complemento } = enderecoData;
    const [result] = await pool.query(
      `INSERT INTO endereco (autuado_id, estado, cidade, cep, bairro, logradouro, numero, complemento) 
       VALUES (?, ?, ?, ?, ?, ?, ?, ?)`,
      [autuado_id, estado, cidade, cep, bairro, logradouro, numero, complemento]
    );
    return { id: result.insertId, ...enderecoData };
  }

  async update(id, enderecoData) {
    const { estado, cidade, cep, bairro, logradouro, numero, complemento } = enderecoData;
    const [result] = await pool.query(
      `UPDATE endereco SET estado = ?, cidade = ?, cep = ?, bairro = ?, 
       logradouro = ?, numero = ?, complemento = ? WHERE id = ?`,
      [estado, cidade, cep, bairro, logradouro, numero, complemento, id]
    );
    return result;
  }

  async delete(id) {
    const [result] = await pool.query('DELETE FROM `endereco` WHERE id = ?', [id]);
    return result;
  }

  async deleteByAutuadoId(autuadoId, connection = pool) {
    const [result] = await connection.query('DELETE FROM `endereco` WHERE autuado_id = ?', [autuadoId]);
    return result;
  }
}

export default new EnderecoRepository();

File: api-python/src/persistence/endereco_repository.py
from mysql.connector import Error

from src.config.connection.connection import DatabaseConnection
from src.schemas.endereco import EnderecoCreate

class EnderecoRepository:
    def buscar_todos(self):
        with DatabaseConnection() as cnx:
            try:
                query = "SELECT * FROM endereco"
                cnx.execute_query(query)
                return cnx.cursor.fetchall()
            except Error as e:
                raise(e)
            
    def buscar_por_id(self, id:int):
        with DatabaseConnection() as cnx:
            try:
                query = "SELECT * FROM endereco WHERE id = %s"
                cnx.execute_query(query, [id])
                return cnx.cursor.fetchall()
            except Error as e:
                raise(e)

    def criar(self, endereco:EnderecoCreate):
        with DatabaseConnection() as cnx:
            try:
                params = [
                    endereco.autuado_id,
                    endereco.estado,
                    endereco.cidade,
                    endereco.cep,
                    endereco.bairro,
                    endereco.logradouro,
                    endereco.complemento,
                    endereco.numero
                ]
                query = """INSERT INTO endereco (
                            autuado_id, estado, cidade, cep, bairro, logradouro, complemento, numero) 
                            VALUES (%s, %s, %s, %s, %s, %s, %s, %s)"""
                cnx.execute_query(query, params)
                cnx.connection.commit()
                return cnx.cursor.lastrowid
            except Error as e:
                raise(e)
            
    def deletar(self, id:int):
        with DatabaseConnection() as cnx:
            try:
                query = "DELETE FROM endereco WHERE id = %s"
                cnx.execute_query(query, [id])
                cnx.connection.commit()
                return cnx.cursor.rowcount
            except Error as e:
                raise(e)